Added BankAccount schema (#37)
fiatconnect/specification: fiatconnect-api.md

      - [8.3.2.2. `MobileMoney`](#8322-mobilemoney)
        * [8.3.2.2.1. `SupportedOperatorEnum`](#83221-supportedoperatorenum)
      - [8.3.2.3. `DuniaWallet`](#8323-duniawallet)

>       - [8.3.2.4. `BankAccount`](#8324-bankaccount)

Reviewer: ```suggestion
      - [8.3.2.4. `IBANNumber`](#8324-ibannumber)
```

}
```

> ####  8.3.2.4. `BankAccount`

Reviewer: How about renaming this to IBANNumber? Since this isn't the only fiat account schema that represents a bank account

Author: Yeah, I am ok with that

Reviewer: ```suggestion
####  8.3.2.4. `IBANNumber`
```

> `BankAccount` is a representation of a bank account that is agnostic to the user's home country. The most important field is the `iban`, which stands for International Bank Account Number. More info on [Wikipedia](https://en.wikipedia.org/wiki/International_Bank_Account_Number) and [World First](https://www.worldfirst.com/uk/help-support/what-is-an-iban-number/).

Reviewer: ```suggestion
`IBANNumber` is a representation of a bank account that is agnostic to the user's home country. The most important field is the `iban`, which stands for International Bank Account Number. More info on [Wikipedia](https://en.wikipedia.org/wiki/International_Bank_Account_Number) and [World First](https://www.worldfirst.com/uk/help-support/what-is-an-iban-number/).
```

```
{
	accountName: `string`,
	institutionName: `string`,

> 	ibanNumber: `string`,

Reviewer: ```suggestion
	iban: `string`,
```

> `BankAccount` is a Fiat Account schema that tends to come up with a bank account representation we can use regardless of the country. Meanwhile, the most important field there is the `iban` that stands for International Bank Account Number. It respects a specific organization that you can find here [**IBAN on WIKIPEDIA**](https://fr.wikipedia.org/wiki/International_Bank_Account_Number) and [**IBAN**](https://www.pricebank.fr/questions-reponses/questions-frequentes/qu-est-ce-que-le-code-iban.html).

Reviewer: ```suggestion
`BankAccount` is a representation of a bank account that is agnostic to the user's home country. The most important field is the `iban`, which stands for International Bank Account Number. More info on [Wikipedia](https://fr.wikipedia.org/wiki/International_Bank_Account_Number) and [pricebank.fr FAQs page](https://www.pricebank.fr/questions-reponses/questions-frequentes/qu-est-ce-que-le-code-iban.html).
```
rewording for clarity

Reviewer: In particular, "tends to" makes it sound like the description only applies part of the time

Reviewer: oh also, since the FiatConnect spec is written in English, I think it would be preferable to have English links for more information-- just noticed these are in french

Author: OK I will figure out a new link to replace this one. Thanks

  bankName?: `string`,
	country: `string`,
	fiatAccountType: `FiatAccountTypeEnum.BankAccount`
}
```

> The `country` field should be a [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) country code. `bankName` was added for verification purpose only (In a case where there maybe a mistake as each bank located in a specific country can only have a unique international identifier).

Reviewer: ```suggestion
The `country` field MUST be a [ISO 3166-1 alpha-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2) country code. `bankName` was added for verification purpose only (In a case where there maybe a mistake as each bank located in a specific country can only have a unique international identifier).
```

Reviewer: there is a logical difference between "should" and "must" to look out for here

Reviewer: Also, what exactly is `bankName` supposed to be? Wouldn't it have to be a machine-readable value to be useful? (e.g. one of a static set of expected values that a CICO provider's server knows what to do with.) And why is it optional? 

If it's just redundant with `iban` because the bank can be inferred from the `iban` number, I vote we leave it out. The client (e.g. a wallet like Valora) can always perform checks like showing the user a logo of the bank associated with that `iban` (assuming this mapping is easy to do). 

Otherwise, we need an enum of valid bank names to be added (see section 8.1 for examples of this).

Author: `bankName` was added for a verification purpose as the IBANNumber contains the Bank code. This is why I did not make it required. But as you stated, the best approach is to remove it and keep only the `iban`.
